src/CredentialSource/AwsNativeSource.php

```diff
@@ -328,6 +328,21 @@ public static function getSigningVarsFromEnv(): ?array
         return null;
     }
 
+    /**
+     * Gets the unique key for caching
+     * For AwsNativeSource the values are:
+     * Imdsv2SessionTokenUrl.SecurityCredentialsUrl.RegionUrl.RegionalCredVerificationUrl
+     *
+     * @return string
+     */
+    public function getCacheKey(): string
+    {
+        return ($this->imdsv2SessionTokenUrl ? $this->imdsv2SessionTokenUrl : '') .
+            '.' . ($this->securityCredentialsUrl ? $this->securityCredentialsUrl : '') .
```

Comment:
PHP trick - when you're dealing with `null`, you can use `??`, and when you're dealing with falsy, you can use `?:` to make this expression shorter
```suggestion
        return ($this->imdsv2SessionTokenUrl ?? '') .
            '.' . ($this->securityCredentialsUrl ?? '') .
```